scripts/quests/sandoria/A_Vicasques_Sermon.lua

```diff
@@ -0,0 +1,140 @@
+require('scripts/globals/zone')
+-----------------------------------
+local ID = require("scripts/zones/Northern_San_dOria/IDs")
+local ronfaureID = require("scripts/zones/East_Ronfaure/IDs")
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.THE_VICASQUE_S_SERMON)
+
+quest.reward =
+{
+    fame = 30,
+    fameArea = xi.quest.fame_area.SANDORIA,
+    item = xi.items.BRASS_RING,
+    title = xi.title.THE_BENEVOLENT_ONE,
+}
+
+quest.sections =
+{
+    -- Speak to Abioleget after completing Waters of the Cheval
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.WATERS_OF_THE_CHEVAL) == QUEST_COMPLETED and
+                player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 2
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Abioleget'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(589)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [589] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- Trade Blue peas to Andelain in East Ronfure
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Abioleget'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(600)
+                    elseif quest:getVar(player, 'Prog') == 0 then
+                        player:showText(npc, ID.text.ABIOLEGET_PEAS, xi.items.BLUE_PEAS, 70)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local gil = trade:getGil()
+                    local count = trade:getItemCount()
+
+                    if gil == 70 and count == 1 then
+                        player:tradeComplete()
+                        player:startEvent(591)
+                        npcUtil.giveItem(player, xi.items.BLUE_PEAS)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [600] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.EAST_RONFAURE] =
+        {
+            ['Andelain'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        player:showText(npc, ronfaureID.text.NO_MORE_NO_LESS, xi.items.BLUE_PEAS)
```

Comment:
These all need to be converted to quest objects:
return quest:messageSpecial()
Alternatively you can use:
return quest:noAction()

Reply:
Gonna need some help with this one later on.